src/WebviewWizardPage.ts

```diff
@@ -6,161 +6,164 @@ import { StandardWizardPageRenderer } from './StandardWizardPageRenderer';
-                      let str : string = this.getRenderer().oneFieldAsString(def, parameters);
-                      templates = templates.concat({id: key + "Field", content: str});
-                  }
-              }
-            }
+  getValidationTemplates(parameters: any, previousParameters: any): Template[] {
+    this.setPageComplete(true);
+    return this.validate(parameters, previousParameters);
+  }
 
-            for( let oneItem of resp.items ) {
-                // Allow users to just put the failed field id here. We add Validation
-                if( !oneItem.template.id.endsWith("Validation")) {
-                    oneItem.template.id = oneItem.template.id + "Validation";
-                }
-                if( oneItem.severity === SEVERITY.ERROR ) {
-                    this.setPageComplete(false);
-                }
-                let img: string = this.severityToImage(oneItem.severity);
-                let clazz = "";
-                if( oneItem.severity == SEVERITY.ERROR)
-                    clazz = "error-message";
-                else if( oneItem.severity == SEVERITY.WARN)
-                    clazz = "warn-message";
-                else
-                    clazz = "info-message";
+  severityToImage(sev: SEVERITY): string {
+    if (sev === SEVERITY.ERROR)
+      return "<i class=\"icon icon__error\"></i>";
+    if (sev === SEVERITY.WARN)
+      return "<i class=\"icon icon__warn\"></i>";
+    if (sev === SEVERITY.INFO)
+      return "<i class=\"icon icon__info\"></i>";
+    return "";
+  }
+  validate(parameters: any, previousParameters: any): Template[] {
+    let templates: Template[] = [];
+    if (this.pageDefinition.validator) {
+      let resp: ValidatorResponse = this.pageDefinition.validator.call(null, parameters, previousParameters);
 
-                let inner = img + (oneItem.template.content ? oneItem.template.content : "&nbsp;");
-                oneItem.template.content = "<div class=\"" + clazz + "\">" + inner + "</div>";
-                templates = templates.concat(oneItem.template);
-            }
+      // If validation has returned any widgets to refresh, we should do that now
+      if (resp.fieldRefresh) {
+        for (let [key, value] of resp.fieldRefresh) {
+          let def: WizardPageFieldDefinition | null = this.findFieldDefinition(key);
+          let currentState: FieldDefinitionState | undefined = this.fieldStateCache.get(key);
+          if (currentState === undefined || currentState === null) {
+            currentState = {};
+            this.fieldStateCache.set(key, currentState);
+          }
+          let stateChanged = false;
+          if (value.hasOwnProperty("enabled")) {
+            stateChanged = currentState.enabled !== value.enabled;
+            currentState.enabled = value.enabled;
+          }
+          if (value.hasOwnProperty("visible")) {
+            stateChanged = stateChanged || currentState.enabled !== value.visible;
```

Comment:
I think you mean currentState.visible instead of `currentState.enabled` ?

Reply:
I did this PTR quickly, thanks for your review, fixed.